File: alphabeticShift/AlphabeticShift.js
/*
	This algorithm replaces each letter in the string with the next charater in the alphabet
*/

const alphabeticShift = inputString => {
	// Declare a const that contains the alphabet
	// Add "a" at the end for instances of "z"
	const alphabet = [..."abcdefghijklmnopqrstuvwxyza"];

	// Convert inputString into an array
	let spStr = [...inputString];
	
	// Loop through each index
	for(let i = 0; i < spStr.length;i++){
		// Grab the next letter in the alphabet array 
		let index = alphabet.indexOf(spStr[i]) + 1; 

		// Assign the current array index the new letter
		spStr[i] = alphabet[index];
	}

	return spStr.join('');
}

console.log(alphabeticShift("aaaabbbccd")); // bbbbcccdde
console.log(alphabeticShift("codesignal")); // dpeftjhobm

File: alphabeticShift/AlphabeticShift.rb
def alphabeticShift(str)
	alphabet = Array['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z','a']
	newStr = ""
	strArr = str.split('')	
	for letter in strArr
		newLetter = alphabet.index(letter) + 1
		newStr += alphabet[alphabet.index(letter) + 1]
	end
	return newStr
end

puts alphabeticShift("aaaabbbccd"); # bbbbcccdde
puts alphabeticShift("codesignal"); # dpeftjhobm
puts alphabeticShift("crazy") # dsbaz